framework/src/utils/MooseUtils.C

```diff
@@ -858,12 +858,24 @@ fileSize(const std::string & filename)
 }
 
 std::string
-realpath(const std::string & path)
+mooseRealpath(const std::string & path)
 {
 #ifdef LIBMESH_HAVE_PETSC
   char dummy[PETSC_MAX_PATH_LEN];
+#if defined(PETSC_HAVE_REALPATH)
+  // If "realpath" is adopted by PETSc and
+  // "path" does not exist, then PETSc will print a misleading message.
+  // [0]PETSC ERROR: Error in external library
+  // [0]PETSC ERROR: realpath()
+  // [0]PETSC ERROR: See https://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
+  // [0]PETSC ERROR: Petsc Release Version 3.13.3, unknown
+  // We here override the misleading message with a better one.
+  if (!realpath(path.c_str(), dummy))
+    mooseError("Failed to get real path for ", path);
```

Comment:
Is this what we want to say? I guess it is still the path but even if you are in the correct directory and spell the input file incorrectly it will report the error. Would something as simple as "input_file_name" not found?

Reply:
Yes, I was trying to make this consistent with the old behavior.

`mooseError("Failed to get real path for ", path);` is there for a while. 


`mooseRealpath` is used for beyond `input_file_name`

Reply:
Or we could simply say:

` mooseError(path, "not found");` 

@dschwen @aeslaughter have any suggestions?

Reply:
That sounds good to me. By doing that it should give the full path to the file. I guess I'm just concerned why include real path? What purpose does real serve here? Is there a fake or imaginary path?

Reply:
It is used to remove symbolic links  

https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/PetscGetRealPath.html

Reply:
OK, I see. the original message actually makes sense since `path` could be a symbolic link . 

I will have to keep as-is